src/main/java/co/cask/gcp/bigquery/BigQuerySink.java

```diff
@@ -121,8 +121,8 @@ public void prepareRun(BatchSinkContext context) throws Exception {
       BigQueryFileFormat.NEWLINE_DELIMITED_JSON,
       TextOutputFormat.class);
 
-    setOutputFormat(context);
     emitLineage(context, fields);
+    setOutputFormat(context);
```

Comment:
same here

Reply:
Nit: can we add more detail
```
Both emitLineage and setOutputFormat internally try to create an external dataset if it does not already exists. We call emitLineage before since it creates the dataset with schema which .
```